internal/error.go

```diff
@@ -119,14 +119,28 @@ Workflow consumers will get an instance of *WorkflowExecutionError. This error w
 */
 
 type (
+	// ApplicationErrorOptions represents a combination of error attributes and additional requests.
+	// All fields are optional, providing flexibility in error customization.
+	// NextRetryInterval is a request from server to override retry interval calculated by the
+	// server according to the RetryPolicy set by the Workflow.
+	// IMPORTANT: NextRetryInterval is meaningful only within the context of errors originating from Activity.
```

Comment:
Nit:  Generally in the Go SDK we would provide documentation for members on the member variables themselves